sdk/typespec/typespec_client_core/src/http/headers/mod.rs

```diff
@@ -273,3 +325,85 @@ impl From<&String> for HeaderValue {
         s.clone().into()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use typespec::error::ErrorKind;
+    use url::Url;
+
+    use super::{FromHeaders, HeaderName, Headers};
+
+    // Just in case we add a ContentLocation struct later, this one is named "ForTest" to indicate it's just here for this test.
+    #[derive(Debug)]
+    struct ContentLocationForTest(Url);
+
+    impl FromHeaders for ContentLocationForTest {
+        type Error = url::ParseError;
+
+        const HEADER_NAMES: &'static [&'static str] = &["content-location"];
```

Comment:
I think this'll lint: consts don't need to declare `'static` as its implied.

Reply:
No longer relevant with the refactor to a function.